diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.c
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.h.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.h
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.h.omv~	2022-08-23 01:06:09.150920016 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.h	2022-08-23 01:06:29.190984061 +0200
@@ -27,6 +27,8 @@
 
 #if defined(NV_DRM_AVAILABLE)
 
+#include <drm/drm_framebuffer.h>
+
 #if defined(NV_DRM_DRMP_H_PRESENT)
 #include <drm/drmP.h>
 #endif
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-drm/nvidia-drm-helper.h.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-drm/nvidia-drm-helper.h
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-drm/nvidia-drm-helper.h.omv~	2022-08-23 01:06:33.403997803 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-drm/nvidia-drm-helper.h	2022-08-23 01:06:43.931032547 +0200
@@ -27,6 +27,8 @@
 
 #if defined(NV_DRM_AVAILABLE)
 
+#include <drm/drm_framebuffer.h>
+
 #if defined(NV_DRM_DRMP_H_PRESENT)
 #include <drm/drmP.h>
 #endif
