diff -uraN NVIDIA-Linux-x86_64-530.41.03/kernel/nvidia-drm/nvidia-drm-drv.c omv-NVIDIA-Linux-x86_64-530.41.03/kernel/nvidia-drm/nvidia-drm-drv.c
--- NVIDIA-Linux-x86_64-530.41.03/kernel/nvidia-drm/nvidia-drm-drv.c	2023-03-16 20:22:14.000000000 +0100
+++ omv-NVIDIA-Linux-x86_64-530.41.03/kernel/nvidia-drm/nvidia-drm-drv.c	2023-05-30 18:54:00.945008547 +0200
@@ -20,6 +20,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
+#include <linux/version.h>
 #include "nvidia-drm-conftest.h" /* NV_DRM_AVAILABLE and NV_DRM_DRM_GEM_H_PRESENT */
 
 #include "nvidia-drm-priv.h"
@@ -1436,7 +1437,11 @@
 
     nv_drm_driver.dumb_create      = nv_drm_dumb_create;
     nv_drm_driver.dumb_map_offset  = nv_drm_dumb_map_offset;
+// Rel. commit "drm: remove dumb_destroy callback" (Christian König, 26 Jan 2023)
+// NB: No resources are leaked, the kernel releases the same resources by default
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
     nv_drm_driver.dumb_destroy     = nv_drm_dumb_destroy;
+#endif
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
 }
diff -uraN NVIDIA-Linux-x86_64-530.41.03/kernel-open/nvidia-drm/nvidia-drm-drv.c omv-NVIDIA-Linux-x86_64-530.41.03/kernel-open/nvidia-drm/nvidia-drm-drv.c
--- NVIDIA-Linux-x86_64-530.41.03/kernel-open/nvidia-drm/nvidia-drm-drv.c	2023-03-16 20:22:15.000000000 +0100
+++ omv-NVIDIA-Linux-x86_64-530.41.03/kernel-open/nvidia-drm/nvidia-drm-drv.c	2023-06-02 20:38:41.467596998 +0200
@@ -20,6 +20,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
+#include <linux/version.h>
 #include "nvidia-drm-conftest.h" /* NV_DRM_AVAILABLE and NV_DRM_DRM_GEM_H_PRESENT */
 
 #include "nvidia-drm-priv.h"
@@ -1436,7 +1437,11 @@
 
     nv_drm_driver.dumb_create      = nv_drm_dumb_create;
     nv_drm_driver.dumb_map_offset  = nv_drm_dumb_map_offset;
+// Rel. commit "drm: remove dumb_destroy callback" (Christian König, 26 Jan 2023)
+// NB: No resources are leaked, the kernel releases the same resources by default
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
     nv_drm_driver.dumb_destroy     = nv_drm_dumb_destroy;
+#endif
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
 }
 

