diff -up NVIDIA-Linux-x86_64-545.29.06/kernel-open/nvidia-drm/nvidia-drm-linux.c.omv~ NVIDIA-Linux-x86_64-545.29.06/kernel-open/nvidia-drm/nvidia-drm-linux.c
--- NVIDIA-Linux-x86_64-545.29.06/kernel-open/nvidia-drm/nvidia-drm-linux.c.omv~	2024-01-24 16:13:00.885339032 +0100
+++ NVIDIA-Linux-x86_64-545.29.06/kernel-open/nvidia-drm/nvidia-drm-linux.c	2024-01-24 16:13:47.337783987 +0100
@@ -47,8 +47,8 @@
 
 MODULE_PARM_DESC(
     modeset,
-    "Enable atomic kernel modesetting (1 = enable, 0 = disable (default))");
-bool nv_drm_modeset_module_param = false;
+    "Enable atomic kernel modesetting (1 = enable (default), 0 = disable)");
+bool nv_drm_modeset_module_param = true;
 module_param_named(modeset, nv_drm_modeset_module_param, bool, 0400);
 
 #if defined(NV_DRM_FBDEV_GENERIC_AVAILABLE)
diff -up NVIDIA-Linux-x86_64-545.29.06/kernel/nvidia-drm/nvidia-drm-linux.c.omv~ NVIDIA-Linux-x86_64-545.29.06/kernel/nvidia-drm/nvidia-drm-linux.c
--- NVIDIA-Linux-x86_64-545.29.06/kernel/nvidia-drm/nvidia-drm-linux.c.omv~	2024-01-24 16:13:54.549853180 +0100
+++ NVIDIA-Linux-x86_64-545.29.06/kernel/nvidia-drm/nvidia-drm-linux.c	2024-01-24 16:14:06.669969528 +0100
@@ -47,8 +47,8 @@
 
 MODULE_PARM_DESC(
     modeset,
-    "Enable atomic kernel modesetting (1 = enable, 0 = disable (default))");
-bool nv_drm_modeset_module_param = false;
+    "Enable atomic kernel modesetting (1 = enable (default), 0 = disable)");
+bool nv_drm_modeset_module_param = true;
 module_param_named(modeset, nv_drm_modeset_module_param, bool, 0400);
 
 #if defined(NV_DRM_FBDEV_GENERIC_AVAILABLE)
