diff -up NVIDIA-Linux-aarch64-525.60.11/kernel/nvidia-uvm/uvm_migrate_pageable.c.omv~ NVIDIA-Linux-aarch64-525.60.11/kernel/nvidia-uvm/uvm_migrate_pageable.c
--- NVIDIA-Linux-aarch64-525.60.11/kernel/nvidia-uvm/uvm_migrate_pageable.c.omv~	2022-11-28 19:03:54.727569245 +0000
+++ NVIDIA-Linux-aarch64-525.60.11/kernel/nvidia-uvm/uvm_migrate_pageable.c	2022-11-28 19:04:02.277587757 +0000
@@ -1001,7 +1001,7 @@ NV_STATUS uvm_migrate_pageable(uvm_migra
     return status;
 }
 
-NV_STATUS uvm_migrate_pageable_init()
+NV_STATUS uvm_migrate_pageable_init(void)
 {
     g_uvm_migrate_vma_state_cache = NV_KMEM_CACHE_CREATE("migrate_vma_state_t", migrate_vma_state_t);
     if (!g_uvm_migrate_vma_state_cache)
@@ -1010,7 +1010,7 @@ NV_STATUS uvm_migrate_pageable_init()
     return NV_OK;
 }
 
-void uvm_migrate_pageable_exit()
+void uvm_migrate_pageable_exit(void)
 {
     kmem_cache_destroy_safe(&g_uvm_migrate_vma_state_cache);
 }
diff -up NVIDIA-Linux-aarch64-525.60.11/kernel-open/nvidia-uvm/uvm_migrate_pageable.c.omv~ NVIDIA-Linux-aarch64-525.60.11/kernel-open/nvidia-uvm/uvm_migrate_pageable.c
--- NVIDIA-Linux-aarch64-525.60.11/kernel-open/nvidia-uvm/uvm_migrate_pageable.c.omv~	2022-11-28 19:19:13.989964938 +0000
+++ NVIDIA-Linux-aarch64-525.60.11/kernel-open/nvidia-uvm/uvm_migrate_pageable.c	2022-11-28 19:19:24.409995696 +0000
@@ -1001,7 +1001,7 @@ NV_STATUS uvm_migrate_pageable(uvm_migra
     return status;
 }
 
-NV_STATUS uvm_migrate_pageable_init()
+NV_STATUS uvm_migrate_pageable_init(void)
 {
     g_uvm_migrate_vma_state_cache = NV_KMEM_CACHE_CREATE("migrate_vma_state_t", migrate_vma_state_t);
     if (!g_uvm_migrate_vma_state_cache)
@@ -1010,7 +1010,7 @@ NV_STATUS uvm_migrate_pageable_init()
     return NV_OK;
 }
 
-void uvm_migrate_pageable_exit()
+void uvm_migrate_pageable_exit(void)
 {
     kmem_cache_destroy_safe(&g_uvm_migrate_vma_state_cache);
 }
