diff -up NVIDIA-Linux-x86_64-545.23.06/kernel-open/nvidia/libspdm_shash.c.omv~ NVIDIA-Linux-x86_64-545.23.06/kernel-open/nvidia/libspdm_shash.c
--- NVIDIA-Linux-x86_64-545.23.06/kernel-open/nvidia/libspdm_shash.c.omv~	2023-11-13 21:40:07.846850346 +0100
+++ NVIDIA-Linux-x86_64-545.23.06/kernel-open/nvidia/libspdm_shash.c	2023-11-13 21:42:17.432123411 +0100
@@ -23,6 +23,15 @@
 
 #include "internal_crypt_lib.h"
 
+#include <linux/version.h>
+#include <linux/crypto.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)
+static inline void *crypto_tfm_ctx_aligned(struct crypto_tfm *tfm)
+{
+	return PTR_ALIGN(tfm, crypto_tfm_ctx_alignment());
+}
+#endif
+
 void *lkca_hash_new(const char* alg_name)
 {
 #ifndef USE_LKCA
diff -up NVIDIA-Linux-x86_64-545.23.06/kernel/nvidia/libspdm_shash.c.omv~ NVIDIA-Linux-x86_64-545.23.06/kernel/nvidia/libspdm_shash.c
--- NVIDIA-Linux-x86_64-545.23.06/kernel/nvidia/libspdm_shash.c.omv~	2023-11-13 21:42:26.994216191 +0100
+++ NVIDIA-Linux-x86_64-545.23.06/kernel/nvidia/libspdm_shash.c	2023-11-13 21:42:34.395287905 +0100
@@ -23,6 +23,15 @@
 
 #include "internal_crypt_lib.h"
 
+#include <linux/version.h>
+#include <linux/crypto.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)
+static inline void *crypto_tfm_ctx_aligned(struct crypto_tfm *tfm)
+{
+	return PTR_ALIGN(tfm, crypto_tfm_ctx_alignment());
+}
+#endif
+
 void *lkca_hash_new(const char* alg_name)
 {
 #ifndef USE_LKCA
