diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/common/inc/nv-linux.h.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/common/inc/nv-linux.h
--- NVIDIA-Linux-x86_64-515.65.01/kernel/common/inc/nv-linux.h.omv~	2022-09-04 20:17:24.353703225 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/common/inc/nv-linux.h	2022-09-04 20:17:42.189469859 +0200
@@ -54,6 +54,8 @@
   #include <generated/compile.h>
 #endif
 
+#include <linux/dma-direct.h>
+
 #include <linux/version.h>
 #include <linux/utsname.h>
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/conftest.sh.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/conftest.sh
--- NVIDIA-Linux-x86_64-515.65.01/kernel/conftest.sh.omv~	2022-09-04 20:10:29.930212025 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/conftest.sh	2022-09-04 20:40:10.626001619 +0200
@@ -6,7 +6,11 @@ PATH="${PATH}:/bin:/sbin:/usr/bin"
 SCRIPTDIR=`dirname $0`
 cd $SCRIPTDIR
 
-CC="$1"
+# FIXME this script is so broken that it falls over additional
+# clang 15 strictness, so we use gcc for the config checks even
+# when building for the clang kernel.
+#CC="$1"
+CC=gcc
 ARCH=$2
 ISYSTEM=`$CC -print-file-name=include 2> /dev/null`
 SOURCES=$3
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/nvlink_linux.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/nvlink_linux.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/nvlink_linux.c.omv~	2022-09-04 20:44:33.145520538 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/nvlink_linux.c	2022-09-04 20:44:38.756429506 +0200
@@ -562,7 +562,7 @@ void nvlink_assert(int cond)
     }
 }
 
-void * nvlink_allocLock()
+void * nvlink_allocLock(void)
 {
     struct semaphore *sema;
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/os-interface.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/os-interface.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/os-interface.c.omv~	2022-09-04 20:40:22.172793731 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia/os-interface.c	2022-09-04 20:40:36.378539553 +0200
@@ -1088,14 +1088,14 @@ void NV_API_CALL os_dbg_breakpoint(void)
 #endif // CONFIG_X86_REMOTE_DEBUG || CONFIG_KGDB || CONFIG_XMON
 }
 
-NvU32 NV_API_CALL os_get_cpu_number()
+NvU32 NV_API_CALL os_get_cpu_number(void)
 {
     NvU32 cpu_id = get_cpu();
     put_cpu();
     return cpu_id;
 }
 
-NvU32 NV_API_CALL os_get_cpu_count()
+NvU32 NV_API_CALL os_get_cpu_count(void)
 {
     return NV_NUM_CPUS();
 }
@@ -1148,7 +1148,7 @@ void NV_API_CALL os_get_screen_info(
 #endif
 }
 
-void NV_API_CALL os_dump_stack()
+void NV_API_CALL os_dump_stack(void)
 {
     dump_stack();
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_gpu_access_counters.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_gpu_access_counters.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_gpu_access_counters.c.omv~	2022-09-04 20:53:46.361119285 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_gpu_access_counters.c	2022-09-04 20:54:01.078906808 +0200
@@ -1535,7 +1535,7 @@ bool uvm_va_space_has_access_counter_mig
     return atomic_read(&va_space_access_counters->params.enable_mimc_migrations);
 }
 
-NV_STATUS uvm_perf_access_counters_init()
+NV_STATUS uvm_perf_access_counters_init(void)
 {
     uvm_perf_module_init("perf_access_counters",
                          UVM_PERF_MODULE_TYPE_ACCESS_COUNTERS,
@@ -1546,7 +1546,7 @@ NV_STATUS uvm_perf_access_counters_init(
     return NV_OK;
 }
 
-void uvm_perf_access_counters_exit()
+void uvm_perf_access_counters_exit(void)
 {
 }
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_lock.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_lock.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_lock.c.omv~	2022-09-04 20:56:04.508138766 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_lock.c	2022-09-04 20:56:15.114987880 +0200
@@ -344,7 +344,7 @@ bool __uvm_check_all_unlocked(uvm_thread
     return false;
 }
 
-bool __uvm_thread_check_all_unlocked()
+bool __uvm_thread_check_all_unlocked(void)
 {
     return __uvm_check_all_unlocked(uvm_thread_context_lock_get());
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_tools.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_tools.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_tools.c.omv~	2022-09-04 20:57:57.018545521 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_tools.c	2022-09-04 20:58:01.936476219 +0200
@@ -2192,7 +2192,7 @@ NV_STATUS uvm_api_tools_get_processor_uu
     return NV_OK;
 }
 
-void uvm_tools_flush_events()
+void uvm_tools_flush_events(void)
 {
     tools_schedule_completed_events();
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_procfs.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_procfs.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_procfs.c.omv~	2022-09-04 20:59:41.158083328 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_procfs.c	2022-09-04 20:59:48.201984802 +0200
@@ -46,7 +46,7 @@ static struct proc_dir_entry *uvm_proc_d
 static struct proc_dir_entry *uvm_proc_gpus;
 static struct proc_dir_entry *uvm_proc_cpu;
 
-NV_STATUS uvm_procfs_init()
+NV_STATUS uvm_procfs_init(void)
 {
     if (!uvm_procfs_is_enabled())
         return NV_OK;
@@ -66,7 +66,7 @@ NV_STATUS uvm_procfs_init()
     return NV_OK;
 }
 
-void uvm_procfs_exit()
+void uvm_procfs_exit(void)
 {
     uvm_procfs_destroy_entry(uvm_proc_dir);
 }
@@ -95,12 +95,12 @@ void uvm_procfs_destroy_entry(struct pro
     procfs_destroy_entry_with_root(entry, entry);
 }
 
-struct proc_dir_entry *uvm_procfs_get_gpu_base_dir()
+struct proc_dir_entry *uvm_procfs_get_gpu_base_dir(void)
 {
     return uvm_proc_gpus;
 }
 
-struct proc_dir_entry *uvm_procfs_get_cpu_base_dir()
+struct proc_dir_entry *uvm_procfs_get_cpu_base_dir(void)
 {
     return uvm_proc_cpu;
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_push.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_push.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_push.c.omv~	2022-09-04 21:01:58.643167586 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_push.c	2022-09-04 21:02:05.927066484 +0200
@@ -270,12 +270,12 @@ NV_STATUS __uvm_push_begin_acquire_on_ch
     return status;
 }
 
-bool uvm_push_info_is_tracking_descriptions()
+bool uvm_push_info_is_tracking_descriptions(void)
 {
     return uvm_debug_enable_push_desc != 0;
 }
 
-bool uvm_push_info_is_tracking_acquires()
+bool uvm_push_info_is_tracking_acquires(void)
 {
     return uvm_debug_enable_push_acquire_info != 0;
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_common.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_common.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_common.c.omv~	2022-09-04 21:09:09.971228710 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_common.c	2022-09-04 21:09:17.783121831 +0200
@@ -34,7 +34,7 @@ static int uvm_debug_prints = UVM_IS_DEB
 module_param(uvm_debug_prints, int, S_IRUGO|S_IWUSR);
 MODULE_PARM_DESC(uvm_debug_prints, "Enable uvm debug prints.");
 
-bool uvm_debug_prints_enabled()
+bool uvm_debug_prints_enabled(void)
 {
     return uvm_debug_prints != 0;
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_thread_context.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_thread_context.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_thread_context.c.omv~	2022-09-04 21:22:40.082708882 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_thread_context.c	2022-09-04 21:22:50.197584259 +0200
@@ -101,7 +101,7 @@ static DEFINE_PER_CPU(uvm_thread_context
 static void thread_context_non_interrupt_remove(uvm_thread_context_t *thread_context,
                                                 uvm_thread_context_table_entry_t *thread_context_entry);
 
-bool uvm_thread_context_wrapper_is_used()
+bool uvm_thread_context_wrapper_is_used(void)
 {
     // The wrapper contains lock information. While uvm_record_lock_X
     // routines are a no-op outside of debug mode, unit tests do invoke their
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_prefetch.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_prefetch.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_prefetch.c.omv~	2022-09-04 21:24:27.974371679 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_prefetch.c	2022-09-04 21:24:34.880285533 +0200
@@ -460,7 +460,7 @@ void uvm_perf_prefetch_unload(uvm_va_spa
     uvm_perf_module_unload(&g_module_prefetch, va_space);
 }
 
-NV_STATUS uvm_perf_prefetch_init()
+NV_STATUS uvm_perf_prefetch_init(void)
 {
     g_uvm_perf_prefetch_enable = uvm_perf_prefetch_enable != 0;
 
@@ -498,7 +498,7 @@ NV_STATUS uvm_perf_prefetch_init()
     return NV_OK;
 }
 
-void uvm_perf_prefetch_exit()
+void uvm_perf_prefetch_exit(void)
 {
     if (!g_uvm_perf_prefetch_enable)
         return;
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_heuristics.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_heuristics.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_heuristics.c.omv~	2022-09-04 21:38:44.434397102 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_heuristics.c	2022-09-04 21:38:51.330307360 +0200
@@ -28,7 +28,7 @@
 #include "uvm_gpu_access_counters.h"
 #include "uvm_va_space.h"
 
-NV_STATUS uvm_perf_heuristics_init()
+NV_STATUS uvm_perf_heuristics_init(void)
 {
     NV_STATUS status;
 
@@ -47,7 +47,7 @@ NV_STATUS uvm_perf_heuristics_init()
     return NV_OK;
 }
 
-void uvm_perf_heuristics_exit()
+void uvm_perf_heuristics_exit(void)
 {
     uvm_perf_access_counters_exit();
     uvm_perf_prefetch_exit();
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_thrashing.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_thrashing.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_thrashing.c.omv~	2022-09-04 21:45:03.577449330 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_perf_thrashing.c	2022-09-04 21:45:15.624291766 +0200
@@ -1949,7 +1949,7 @@ NV_STATUS uvm_perf_thrashing_register_gp
     return NV_OK;
 }
 
-NV_STATUS uvm_perf_thrashing_init()
+NV_STATUS uvm_perf_thrashing_init(void)
 {
     NV_STATUS status;
 
@@ -2008,7 +2008,7 @@ error:
     return status;
 }
 
-void uvm_perf_thrashing_exit()
+void uvm_perf_thrashing_exit(void)
 {
     cpu_thrashing_stats_exit();
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_migrate.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_migrate.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_migrate.c.omv~	2022-09-04 22:00:38.484624154 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel/nvidia-uvm/uvm_migrate.c	2022-09-04 22:00:44.031556062 +0200
@@ -791,7 +791,7 @@ static NV_STATUS semaphore_release(NvU64
     return semaphore_release_from_gpu(gpu, semaphore_pool, semaphore_address, semaphore_payload, tracker_ptr);
 }
 
-NV_STATUS uvm_migrate_init()
+NV_STATUS uvm_migrate_init(void)
 {
     NV_STATUS status = uvm_migrate_pageable_init();
     if (status != NV_OK)
@@ -817,7 +817,7 @@ NV_STATUS uvm_migrate_init()
     return NV_OK;
 }
 
-void uvm_migrate_exit()
+void uvm_migrate_exit(void)
 {
     uvm_migrate_pageable_exit();
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/nvlink_linux.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/nvlink_linux.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/nvlink_linux.c.omv~	2022-09-04 22:25:56.257206855 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/nvlink_linux.c	2022-09-04 22:25:59.268161034 +0200
@@ -562,7 +562,7 @@ void nvlink_assert(int cond)
     }
 }
 
-void * nvlink_allocLock()
+void * nvlink_allocLock(void)
 {
     struct semaphore *sema;
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/os-interface.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/os-interface.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/os-interface.c.omv~	2022-09-04 22:25:37.967485941 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia/os-interface.c	2022-09-04 22:25:50.598293064 +0200
@@ -1088,14 +1088,14 @@ void NV_API_CALL os_dbg_breakpoint(void)
 #endif // CONFIG_X86_REMOTE_DEBUG || CONFIG_KGDB || CONFIG_XMON
 }
 
-NvU32 NV_API_CALL os_get_cpu_number()
+NvU32 NV_API_CALL os_get_cpu_number(void)
 {
     NvU32 cpu_id = get_cpu();
     put_cpu();
     return cpu_id;
 }
 
-NvU32 NV_API_CALL os_get_cpu_count()
+NvU32 NV_API_CALL os_get_cpu_count(void)
 {
     return NV_NUM_CPUS();
 }
@@ -1148,7 +1148,7 @@ void NV_API_CALL os_get_screen_info(
 #endif
 }
 
-void NV_API_CALL os_dump_stack()
+void NV_API_CALL os_dump_stack(void)
 {
     dump_stack();
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_procfs.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_procfs.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_procfs.c.omv~	2022-09-04 22:26:23.575792428 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_procfs.c	2022-09-04 22:26:31.732669239 +0200
@@ -46,7 +46,7 @@ static struct proc_dir_entry *uvm_proc_d
 static struct proc_dir_entry *uvm_proc_gpus;
 static struct proc_dir_entry *uvm_proc_cpu;
 
-NV_STATUS uvm_procfs_init()
+NV_STATUS uvm_procfs_init(void)
 {
     if (!uvm_procfs_is_enabled())
         return NV_OK;
@@ -66,7 +66,7 @@ NV_STATUS uvm_procfs_init()
     return NV_OK;
 }
 
-void uvm_procfs_exit()
+void uvm_procfs_exit(void)
 {
     uvm_procfs_destroy_entry(uvm_proc_dir);
 }
@@ -95,12 +95,12 @@ void uvm_procfs_destroy_entry(struct pro
     procfs_destroy_entry_with_root(entry, entry);
 }
 
-struct proc_dir_entry *uvm_procfs_get_gpu_base_dir()
+struct proc_dir_entry *uvm_procfs_get_gpu_base_dir(void)
 {
     return uvm_proc_gpus;
 }
 
-struct proc_dir_entry *uvm_procfs_get_cpu_base_dir()
+struct proc_dir_entry *uvm_procfs_get_cpu_base_dir(void)
 {
     return uvm_proc_cpu;
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_gpu_access_counters.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_gpu_access_counters.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_gpu_access_counters.c.omv~	2022-09-04 22:48:01.565744174 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_gpu_access_counters.c	2022-09-04 22:48:07.344666949 +0200
@@ -1535,7 +1535,7 @@ bool uvm_va_space_has_access_counter_mig
     return atomic_read(&va_space_access_counters->params.enable_mimc_migrations);
 }
 
-NV_STATUS uvm_perf_access_counters_init()
+NV_STATUS uvm_perf_access_counters_init(void)
 {
     uvm_perf_module_init("perf_access_counters",
                          UVM_PERF_MODULE_TYPE_ACCESS_COUNTERS,
@@ -1546,7 +1546,7 @@ NV_STATUS uvm_perf_access_counters_init(
     return NV_OK;
 }
 
-void uvm_perf_access_counters_exit()
+void uvm_perf_access_counters_exit(void)
 {
 }
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_tools.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_tools.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_tools.c.omv~	2022-09-04 22:59:45.365120610 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_tools.c	2022-09-04 22:59:51.312991188 +0200
@@ -2192,7 +2192,7 @@ NV_STATUS uvm_api_tools_get_processor_uu
     return NV_OK;
 }
 
-void uvm_tools_flush_events()
+void uvm_tools_flush_events(void)
 {
     tools_schedule_completed_events();
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_lock.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_lock.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_lock.c.omv~	2022-09-04 23:12:24.983051486 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_lock.c	2022-09-04 23:12:33.143916875 +0200
@@ -344,7 +344,7 @@ bool __uvm_check_all_unlocked(uvm_thread
     return false;
 }
 
-bool __uvm_thread_check_all_unlocked()
+bool __uvm_thread_check_all_unlocked(void)
 {
     return __uvm_check_all_unlocked(uvm_thread_context_lock_get());
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_common.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_common.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_common.c.omv~	2022-09-04 23:14:05.952399192 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_common.c	2022-09-04 23:14:12.729289257 +0200
@@ -34,7 +34,7 @@ static int uvm_debug_prints = UVM_IS_DEB
 module_param(uvm_debug_prints, int, S_IRUGO|S_IWUSR);
 MODULE_PARM_DESC(uvm_debug_prints, "Enable uvm debug prints.");
 
-bool uvm_debug_prints_enabled()
+bool uvm_debug_prints_enabled(void)
 {
     return uvm_debug_prints != 0;
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_push.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_push.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_push.c.omv~	2022-09-04 23:17:10.144447264 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_push.c	2022-09-04 23:17:15.741358605 +0200
@@ -270,12 +270,12 @@ NV_STATUS __uvm_push_begin_acquire_on_ch
     return status;
 }
 
-bool uvm_push_info_is_tracking_descriptions()
+bool uvm_push_info_is_tracking_descriptions(void)
 {
     return uvm_debug_enable_push_desc != 0;
 }
 
-bool uvm_push_info_is_tracking_acquires()
+bool uvm_push_info_is_tracking_acquires(void)
 {
     return uvm_debug_enable_push_acquire_info != 0;
 }
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_thread_context.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_thread_context.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_thread_context.c.omv~	2022-09-04 23:19:46.899982671 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_thread_context.c	2022-09-04 23:19:52.757891243 +0200
@@ -101,7 +101,7 @@ static DEFINE_PER_CPU(uvm_thread_context
 static void thread_context_non_interrupt_remove(uvm_thread_context_t *thread_context,
                                                 uvm_thread_context_table_entry_t *thread_context_entry);
 
-bool uvm_thread_context_wrapper_is_used()
+bool uvm_thread_context_wrapper_is_used(void)
 {
     // The wrapper contains lock information. While uvm_record_lock_X
     // routines are a no-op outside of debug mode, unit tests do invoke their
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_thrashing.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_thrashing.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_thrashing.c.omv~	2022-09-04 23:22:50.676132854 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_thrashing.c	2022-09-04 23:22:56.751039230 +0200
@@ -1949,7 +1949,7 @@ NV_STATUS uvm_perf_thrashing_register_gp
     return NV_OK;
 }
 
-NV_STATUS uvm_perf_thrashing_init()
+NV_STATUS uvm_perf_thrashing_init(void)
 {
     NV_STATUS status;
 
@@ -2008,7 +2008,7 @@ error:
     return status;
 }
 
-void uvm_perf_thrashing_exit()
+void uvm_perf_thrashing_exit(void)
 {
     cpu_thrashing_stats_exit();
 
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_prefetch.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_prefetch.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_prefetch.c.omv~	2022-09-04 23:27:01.525290148 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_prefetch.c	2022-09-04 23:27:11.844132942 +0200
@@ -460,7 +460,7 @@ void uvm_perf_prefetch_unload(uvm_va_spa
     uvm_perf_module_unload(&g_module_prefetch, va_space);
 }
 
-NV_STATUS uvm_perf_prefetch_init()
+NV_STATUS uvm_perf_prefetch_init(void)
 {
     g_uvm_perf_prefetch_enable = uvm_perf_prefetch_enable != 0;
 
@@ -498,7 +498,7 @@ NV_STATUS uvm_perf_prefetch_init()
     return NV_OK;
 }
 
-void uvm_perf_prefetch_exit()
+void uvm_perf_prefetch_exit(void)
 {
     if (!g_uvm_perf_prefetch_enable)
         return;
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_heuristics.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_heuristics.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_heuristics.c.omv~	2022-09-04 23:30:42.387042006 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_perf_heuristics.c	2022-09-04 23:30:48.384983713 +0200
@@ -28,7 +28,7 @@
 #include "uvm_gpu_access_counters.h"
 #include "uvm_va_space.h"
 
-NV_STATUS uvm_perf_heuristics_init()
+NV_STATUS uvm_perf_heuristics_init(void)
 {
     NV_STATUS status;
 
@@ -47,7 +47,7 @@ NV_STATUS uvm_perf_heuristics_init()
     return NV_OK;
 }
 
-void uvm_perf_heuristics_exit()
+void uvm_perf_heuristics_exit(void)
 {
     uvm_perf_access_counters_exit();
     uvm_perf_prefetch_exit();
diff -up NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate.c.omv~ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate.c
--- NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate.c.omv~	2022-09-04 23:33:44.733144849 +0200
+++ NVIDIA-Linux-x86_64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate.c	2022-09-04 23:33:53.430048641 +0200
@@ -791,7 +791,7 @@ static NV_STATUS semaphore_release(NvU64
     return semaphore_release_from_gpu(gpu, semaphore_pool, semaphore_address, semaphore_payload, tracker_ptr);
 }
 
-NV_STATUS uvm_migrate_init()
+NV_STATUS uvm_migrate_init(void)
 {
     NV_STATUS status = uvm_migrate_pageable_init();
     if (status != NV_OK)
@@ -817,7 +817,7 @@ NV_STATUS uvm_migrate_init()
     return NV_OK;
 }
 
-void uvm_migrate_exit()
+void uvm_migrate_exit(void)
 {
     uvm_migrate_pageable_exit();
 }
diff -up NVIDIA-Linux-aarch64-515.65.01/kernel/nvidia-uvm/uvm_migrate_pageable.c.omv~ NVIDIA-Linux-aarch64-515.65.01/kernel/nvidia-uvm/uvm_migrate_pageable.c
--- NVIDIA-Linux-aarch64-515.65.01/kernel/nvidia-uvm/uvm_migrate_pageable.c.omv~	2022-09-05 01:11:01.465583188 +0100
+++ NVIDIA-Linux-aarch64-515.65.01/kernel/nvidia-uvm/uvm_migrate_pageable.c	2022-09-05 01:11:13.985596751 +0100
@@ -1001,7 +1001,7 @@ NV_STATUS uvm_migrate_pageable(uvm_migra
     return status;
 }
 
-NV_STATUS uvm_migrate_pageable_init()
+NV_STATUS uvm_migrate_pageable_init(void)
 {
     g_uvm_migrate_vma_state_cache = NV_KMEM_CACHE_CREATE("migrate_vma_state_t", migrate_vma_state_t);
     if (!g_uvm_migrate_vma_state_cache)
@@ -1010,7 +1010,7 @@ NV_STATUS uvm_migrate_pageable_init()
     return NV_OK;
 }
 
-void uvm_migrate_pageable_exit()
+void uvm_migrate_pageable_exit(void)
 {
     kmem_cache_destroy_safe(&g_uvm_migrate_vma_state_cache);
 }
diff -up NVIDIA-Linux-aarch64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate_pageable.c.omv~ NVIDIA-Linux-aarch64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate_pageable.c
--- NVIDIA-Linux-aarch64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate_pageable.c.omv~	2022-09-05 01:17:57.456031672 +0100
+++ NVIDIA-Linux-aarch64-515.65.01/kernel-open/nvidia-uvm/uvm_migrate_pageable.c	2022-09-05 01:18:03.416038073 +0100
@@ -1001,7 +1001,7 @@ NV_STATUS uvm_migrate_pageable(uvm_migra
     return status;
 }
 
-NV_STATUS uvm_migrate_pageable_init()
+NV_STATUS uvm_migrate_pageable_init(void)
 {
     g_uvm_migrate_vma_state_cache = NV_KMEM_CACHE_CREATE("migrate_vma_state_t", migrate_vma_state_t);
     if (!g_uvm_migrate_vma_state_cache)
@@ -1010,7 +1010,7 @@ NV_STATUS uvm_migrate_pageable_init()
     return NV_OK;
 }
 
-void uvm_migrate_pageable_exit()
+void uvm_migrate_pageable_exit(void)
 {
     kmem_cache_destroy_safe(&g_uvm_migrate_vma_state_cache);
 }
